### PR TITLE
Reduce the possibility of OOM removing since the event contente is retrieved through a DataSupplier.

### PR DESCRIPTION
**Pull request by saratry**

### CODE REVIEW

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/SerializedEvent.java [hunk 1]
```diff
@@ -33,34 +33,41 @@
...
+        this.dataSupplier = () -> eventFromFile;
     }
 
     public SerializedEvent(InputStream event) {
-        try {
-            this.serializedData = toByteArray(event);
-        } catch (IOException e) {
-            throw new MessagingPlatformException(ErrorCode.DATAFILE_READ_ERROR, e.getMessage(), e);
-        }
+        this.dataSupplier = () -> {
```
Comment (MGathier) [inline axonserver/src/main/java/io/axoniq/axonserver/localstorage/SerializedEvent.java:54]: This constructor is used while receiving events from the client. If you delay the reading of the stream until it is used, the stream may be closed already by gRPC.

Comment (MGathier) [reply axonserver/src/main/java/io/axoniq/axonserver/localstorage/SerializedEvent.java:54]: also, the input stream can only be read once, so the second time the dataSupplier is called it fails.

**Review by MGathier — CHANGES_REQUESTED** [review]
All together I don't see that much value in this change. The most common case where the serialized event would remain in the Axon Server for a slightly longer period of time is during the append, in which case it is using the constructor with the InputStream. This constructor must be changed to read the stream immediately, otherwise it may be gone by the time we try to read it. And even if not, the message would still be in memory.

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/SerializedEvent.java [hunk 2]
```diff
@@ -120,10 +133,16 @@ public String getType() {
...
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        SerializedEvent that = (SerializedEvent) o;
+        return Arrays.equals(serializedData(), that.serializedData());
     }
```
Comment (bert-laverman) [inline axonserver/src/main/java/io/axoniq/axonserver/localstorage/SerializedEvent.java:146]: So every time `equals()` is called, the event is deserialized? On both sides? Does that really save memory?

**Review by bert-laverman — CHANGES_REQUESTED** [review]
`serializedData()` is going to be called a _lot_.

**Pull request closed without merging**